core/domain/exp_jobs_one_off.py

```diff
@@ -57,6 +60,32 @@ def map(exploration_model):
     def reduce(exp_id, list_of_exps):
         pass
 
+class ExpSummariesContributorsOneOffJob(jobs.BaseMapReduceJobManager):
+    """ One-off job that finds the user id's of the contributors
+    (defined as any human who has made a positive (i.e. not just
+    a reversion) for each exploration.
+    """
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationSnapshotMetadataModel]
+
+    @staticmethod
+    def map(item):
+        _VERSION_DELIMITER = '-'
```

Comment:
(1) Not sure. Could you raise an exception artificially or print something to get a better sense of what's happening? I think this is something that's more likely to be solved by debugging than by googling. You'll want to make sure that self.id is a string.
(2) Yes -- I suggest putting it in both base models.